FluidNC/src/Maslow/Maslow.cpp

```diff
@@ -764,10 +787,22 @@ bool Maslow_::take_measurement(int waypoint, int dir, int run) {
         //once both belts are pulled, take a measurement
         if (BR_tight && BL_tight) {
             //take measurement and record it to the calibration data array
-            calibration_data[0][waypoint] = axisTL.getPosition() + _beltEndExtension + _armLength;
-            calibration_data[1][waypoint] = axisTR.getPosition() + _beltEndExtension + _armLength;
-            calibration_data[2][waypoint] = axisBL.getPosition() + _beltEndExtension + _armLength;
-            calibration_data[3][waypoint] = axisBR.getPosition() + _beltEndExtension + _armLength;
+            b = z + tlZ;
```

Comment:
I'm trying to build this locally using platformio and it looks like `z` hasn't been defined at this point in the code.

Reply:
ditto line 895

Reply:
It looks like that bit of code just relates to the projection of the measurements onto the XY plane which seems like it should be in it's own pull request so I reverted that commit and it should build now

Reply:
I missed that. If we just remove z from the calculation it will require the Z axis be all the way down (as current calibration instructions say to do)

The UI has a projectMeasurements() function that isn't called anywhere. Rather than modifying the UI to call it, I think it would be better to do the projection here.

We have people working on different calibration calculations, different UI options, and people are looking for additional flexibility in the belt lengths. These include:
extensions
clamps on the belts (so you don't have to completely remove the router from the frame)
the ability to calibrate without the Z axis all the way down (which I thought adding the Z height here would do)
etc.

So I think it's better to report the projected lengths to the UI and calibration routines rather than having them all implement the projection and have to all be modified along with the first commit that does the projected length -> needed belt calculation. better to keep such changes in one PR in one codebase